### PR TITLE
Enhance personalization and the ITagProvider to allow access to all device attributes

### PR DESCRIPTION
**Pull request by jrmihalick**
### Summary
- Adds `IDeviceAttributes` interface in order to keep `ITagProvider` in `openpos-persist` project.  `DeviceModel` implements the `IDeviceAttributes` interface so that the device properties can be provided to `ITagProvider` implementations.
- This is in preparation for dynamically setting the `device_type` tag during Petco personalization
- Clean up a couple of lint warnings

### CODE REVIEW

File: openpos-devices/src/main/java/org/jumpmind/pos/devices/model/DeviceModel.java [hunk 1]
```diff
@@ -59,6 +58,8 @@
     @Builder.Default
     private String deviceMode = DEVICE_MODE_DEFAULT;
 
+    private List<DeviceParamModel> deviceParamModels;
```
Comment (jrmihalick) [inline openpos-devices/src/main/java/org/jumpmind/pos/devices/model/DeviceModel.java:61]: I just moved this from the bottom of the class def

**Comment by jrmihalick** [comment]
@chenson42, @mmichalek  : Could I get a review on this since it will eventually be coming to master?

File: openpos-persist/src/main/java/org/jumpmind/pos/persist/ITagProvider.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package org.jumpmind.pos.persist;
 
-import org.springframework.core.env.Environment;
+import org.jumpmind.pos.util.model.IDeviceAttributes;
 
 
 public interface ITagProvider {
 
-    public String getTagValue(String deviceId, String businessUnitId, String tagName);
+    String getTagValue(IDeviceAttributes deviceModel, String tagName);
```
Comment (chenson42) [inline openpos-persist/src/main/java/org/jumpmind/pos/persist/ITagProvider.java:8]: maybe change deviceModel to attributes

Comment (jrmihalick) [reply openpos-persist/src/main/java/org/jumpmind/pos/persist/ITagProvider.java:8]: Whoops, yeah

**Review by chenson42 — APPROVED** [review]

**Pull request merged**